mongodbatlas/data_source_mongodbatlas_data_lake.go

```diff
@@ -79,6 +79,7 @@ func dataSourceMongoDBAtlasDataLake() *schema.Resource {
 			"storage_databases": schemaDataLakesDatabases(),
 			"storage_stores":    schemaDataLakesStores(),
 		},
+		DeprecationMessage: "this resource is deprecated, and will be removed in v1.11.0 release, please transition as soon as possible to mongodbatlas_federated_database_instance",
```

Comment:
```suggestion
		DeprecationMessage: "this resource is deprecated, and will be removed in v1.12.0 release, please transition as soon as possible to mongodbatlas_federated_database_instance",
```
since we are now releasing this resource in v1.10.0, we need to give our users 2 full version before removing. bumping up removal to v1.12.0

Reply:
updated